File: llvm/OverflowTool/tests/regression/annotation_test.c
// RUN: clang -c -O0 -Xclang -disable-O0-optnone -S -emit-llvm %s -o %t1.ll
// RUN: %bindir/run-oft-manual-annot-sel-print %t1.ll 2>&1 | grep -v %t1.ll | wc -l | cmp <(echo 1)

#include <stdio.h>

extern void oft_mark_(void *);

int main() {
    int a_number = 42;
    oft_mark_(&a_number);

    printf("Local num is %d\n", a_number * 7);
    return 0;
}

File: llvm/OverflowTool/tests/regression/annotation_test.f90
! RUN: flang -g -c -O0 -Xclang -disable-O0-optnone -S -emit-llvm %s -o %t1.ll
! RUN: opt -load-pass-plugin %libdir/libLLVMOverflowToolPass.so -aa-pipeline='basic-aa' -passes='oft-overflow-instrumentation' -S -o %t1.instrumented.ll %t1.ll 2>%t1.passout.ll
! RUN: %bindir/check_marked_lines %t1.passout.ll 10 

program annotation_test
    implicit none
    external oft_mark
    integer val
    call oft_mark(val)
    print*, 'Number is', val * 3 !mul should be marked
end
